app/src/main/java/net/programmierecke/radiodroid2/service/RadioDroidBrowser.java

```diff
@@ -169,9 +182,27 @@ public RadioDroidBrowser(RadioDroidApp radioDroidApp) {
         this.radioDroidApp = radioDroidApp;
     }
 
-    @Nullable
+    public static final String CONTENT_STYLE_SUPPORTED = "android.media.browse.CONTENT_STYLE_SUPPORTED";
+    public static final String CONTENT_STYLE_PLAYABLE_HINT = "android.media.browse.CONTENT_STYLE_PLAYABLE_HINT";
+    public static final String CONTENT_STYLE_BROWSABLE_HINT = "android.media.browse.CONTENT_STYLE_BROWSABLE_HINT";
+    public static final int CONTENT_STYLE_LIST_ITEM_HINT_VALUE = 1;
+    public static final int CONTENT_STYLE_GRID_ITEM_HINT_VALUE = 2;
+
+   @Nullable
     public MediaBrowserServiceCompat.BrowserRoot onGetRoot(@NonNull String clientPackageName, int clientUid, @Nullable Bundle rootHints) {
-        return new MediaBrowserServiceCompat.BrowserRoot(MEDIA_ID_ROOT, null);
+       SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(radioDroidApp.getApplicationContext().getApplicationContext());
+       showIconsInBrowser = sharedPref.getBoolean("show_station_icons_in_hud_browser", false);
+       Bundle extras = new Bundle();
+       extras.putInt(CONTENT_STYLE_BROWSABLE_HINT, CONTENT_STYLE_LIST_ITEM_HINT_VALUE);
+       if (showIconsInBrowser) {
+           Log.d(TAG, "Settimg grid style for playables");
+           extras.putInt(CONTENT_STYLE_PLAYABLE_HINT, CONTENT_STYLE_GRID_ITEM_HINT_VALUE);
+       } else {
+           Log.d(TAG, "Settimg list style for playables");
```

Comment:
Did you mean "setting"?

Reply:
Yes, thanks! This will need one more setting, anyway. In the grid view, the icons are huge....